Scripts/AppCenter/build-appcenter-ios-framework.sh

```diff
@@ -41,8 +44,32 @@ cp -f "${SRCROOT}/${FMK_NAME}/Support/iOS.modulemap" "${INSTALL_DIR}/Modules/mod
 # Copies the headers and resources files to the final product folder.
 cp -R "${SRCROOT}/${WRK_DIR}/Release-iphoneos/include/${FMK_NAME}/" "${INSTALL_DIR}/Headers/"
 
-# Uses the Lipo Tool to merge both binary files (i386 + armv6/armv7/armv7s/arm64) into one Universal final product.
-lipo -create "${DEVICE_DIR}/lib${FMK_NAME}.a" "${SIMULATOR_DIR}/lib${FMK_NAME}.a" -output "${INSTALL_DIR}/${FMK_NAME}"
+# Create the arm64e slice in Xcode 10.1 and lipo it with the device binary that was created with oldest supported Xcode version.
+# Move binary that was create with old Xcode to temp location.
+LIB_IPHONEOS_FINAL="${DEVICE_DIR}/lib${FMK_NAME}.a"
+if [ -z "$MS_ARM64E_XCODE_PATH" ] || [ ! -d "$MS_ARM64E_XCODE_PATH" ] ; then
```

Comment:
`[ -z "$MS_ARM64E_XCODE_PATH" ] || [ ! -d "$MS_ARM64E_XCODE_PATH" ]` condition is used twice in the file. You can declare a local variable or you can combine two `if` statements.
This script shows it like
```
if (condition) {
  // Print error message
} else {
  // Building arm64e slice
}
if (condition) {
  // Use lagacy Xcode to generate fat binary
} else {
  // Use arm64e Xcode to generate fat binary
}
```
You can do it like
```
if (condition) {
  // Print error message
  // Use lagacy Xcode to generate fat binary
} else {
  // Building arm64e slice
  // Use arm64e Xcode to generate fat binary
}
```

Reply:
I don't know if you copied from the script somewhere @jaelim-ms, but if you did then it's *`legacy`.  Also FAT should be capitalized

Reply:
@jaelim-ms I think we should do general script optimizations in a bug since we have other things - like tons of duplicated code. We could do your suggestion there. Does that make sense?

Reply:
What I meant was that the code between line 50 and 72 could be written like below.
```
if [ -z "$MS_ARM64E_XCODE_PATH" ] || [ ! -d "$MS_ARM64E_XCODE_PATH" ] ; then
  echo "Environment variable MS_ARM64E_XCODE_PATH not set or not a valid path."
  # Uses the Lipo tool to merge both binary files (i386/x86_64 + armv7/armv7s/arm64/arm64e) into one universal final product.
  echo "Use legacy Xcode and lipo -create the fat binary."
  lipo -create "${LIB_IPHONEOS_FINAL}" "${SIMULATOR_DIR}/lib${FMK_NAME}.a" -output 
  "${INSTALL_DIR}/${FMK_NAME}"
else
  echo "Building the arm64e slice."
  LIB_IPHONEOS_TEMP_DIR="${DEVICE_DIR}/temp"
  mkdir -p "${LIB_IPHONEOS_TEMP_DIR}"
  mv "${DEVICE_DIR}/lib${FMK_NAME}.a" "${LIB_IPHONEOS_TEMP_DIR}/lib${FMK_NAME}.a"

  # Build with the Xcode version that supports arm64e.
  env DEVELOPER_DIR="${MS_ARM64E_XCODE_PATH}" /usr/bin/xcodebuild ARCHS="arm64e" - project "${FMK_NAME}.xcodeproj" -configuration "Release" -target "${TGT_NAME}"

  # Lipo the binaries that were built from various Xcode versions.
  env DEVELOPER_DIR="${MS_ARM64E_XCODE_PATH}" /usr/bin/lipo -create 
  "${LIB_IPHONEOS_TEMP_DIR}/lib${FMK_NAME}.a" "${LIB_IPHONEOS_FINAL}" -output 
  "${LIB_IPHONEOS_FINAL}"

  # Uses the Lipo tool to merge both binary files (i386/x86_64 + armv7/armv7s/arm64/arm64e) into one universal final product.
  echo "Use arm64e Xcode and lipo -create the fat binary."
  env DEVELOPER_DIR="$MS_ARM64E_XCODE_PATH" /usr/bin/lipo -create 
  "${LIB_IPHONEOS_FINAL}" "${SIMULATOR_DIR}/lib${FMK_NAME}.a" -output 
  "${INSTALL_DIR}/${FMK_NAME}"
fi
```

Reply:
Made a [bug](https://msmobilecenter.visualstudio.com/Mobile-Center/_workitems/edit/53938)

Reply:
So we've actually cleaned this up a bit this morning. The bug that @achocron created is still valid because of duplicate code across the build scripts.